test/run_test.py

```diff
@@ -228,12 +228,19 @@ def test_distributed(executable, test_module, test_directory, options):
             os.environ['BACKEND'] = backend
             os.environ['INIT_METHOD'] = 'env://'
             os.environ.update(env_vars)
-            if with_init_file:
-                if test_module == "test_distributed":
-                    init_method = 'file://{}/'.format(tmp_dir)
-                else:
-                    init_method = 'file://{}/shared_init_file'.format(tmp_dir)
+
+            # Tests are flaky if using TCP store in test_distributed.py,
+            # as ports could be racy, see flaky test
+            # https://github.com/pytorch/pytorch/issues/30676.
+            # Since test_distributed.py does not have TCP store related tests,
+            # making test_distributed to use file store only.
```

Comment:
We should pass in 0 as the port to avoid port conflicts. See https://github.com/pytorch/pytorch/pull/31674 for an example.

Reply:
If its a large refactor to do this and we'd like to get this PR in to fix a flaky test, lets create a gh issue to fix TCPStore port conflicts across our unit tests and tag it with better-engineering.

Reply:
I see, in this case we can still test TCPStore, but passing master_port as 0 to avoid port conflicts, will change it